src/main/java/org/opensearch/neuralsearch/query/AgenticSearchQueryBuilder.java

```diff
@@ -171,24 +181,31 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) throws
 
         // Get index mapping from cluster state
         String indexMapping = NeuralSearchClusterUtil.instance().getIndexMapping(coordinatorContext);
+        String indexMappingJson = gson.toJson(indexMapping);
 
         // Execute agent at coordinator level to avoid multiple calls per shard
         CompletableFuture<String> future = new CompletableFuture<>();
         Map<String, String> parameters = new HashMap<>();
         parameters.put("query_text", queryText);
-        parameters.put("index_mapping", indexMapping);
+        parameters.put("index_mapping", indexMappingJson);
         if (queryFields != null && !queryFields.isEmpty()) {
             parameters.put("query_fields", String.join(",", queryFields));
         }
 
-//        ML_CLIENT.executeAgent(agentId, parameters, ActionListener.wrap(future::complete, future::completeExceptionally));
+        ML_CLIENT.executeAgent(agentId, parameters, ActionListener.wrap(future::complete, future::completeExceptionally));
 
         try {
-            String dslQueryString = future.get();
-            // String dslQueryString = "{" + "\"match\": {" + "\"passage_text\": {" + "\"query\": \"science\"" + "}" + "}" + "}";
+            String agentResponse = future.get();
```

Comment:
when the prediction return no result, the string is "null" or "", we should return `multi-match` as default. That would give a much better user experience. Because if prediction result is null, in this case, it would throw out parsing exception. 

for example, the prediction result is null or empty. we should return a multi-match query
`
{
  "query": {
    "multi_match" : {
      "query":    "help me find a shoes", // the question from query_text
      "fields": [ "title", "*_name" ] // the fields from query_fields
    }
  }
}`

Reply:
I can make query planning tool to return multi_match query when model prediction is null or empty. 

But on the query side, I still think some validations should happen on the query side to double check the string getting from the agent response is a proper query string.

Reply:
Sure, will add more validation on the query side

Reply:
Added validation if the generated query is not wrapped inside a "query" field